tenant-option-management/tenant-option-management.service.ts

```diff
@@ -62,12 +64,35 @@ export class TenantOptionManagementService {
     return this.tenantOption.update(option);
   }
 
-  getAllOptions() {
-    return this.tenantOption
-      .list({
-        pageSize: 2000,
-      })
-      .then((res) => res.data.map((o) => ({ id: `${o.category}-${o.key}`, value: o.value })));
+  async getAllOptions(): Promise<{ id: string; value: string }[]> {
+    try {
+      const tenantOptions: ITenantOption[] = [];
+      const response = await this.tenantOption.list({
+        pageSize: this.MAX_PAGE_SIZE,
+        withTotalPages: true,
+      });
+
+      tenantOptions.push(...response.data);
+
+      for (
+        let currentPage = response.paging.currentPage + 1;
+        currentPage <= response.paging.totalPages;
+        currentPage++
+      ) {
+        const { data } = await this.tenantOption.list({
+          pageSize: this.MAX_PAGE_SIZE,
+          currentPage: currentPage,
+        });
+
+        tenantOptions.push(...data);
+      }
+
+      return tenantOptions.map((o) => ({ id: `${o.category}-${o.key}`, value: o.value }));
+    } catch (error) {
+      console.error(error);
+
+      return undefined;
```

Comment:
Returning undefined while not handling that case in the reload method of the tenant-option.component.ts will lead to an NPE when trying to call find on undefined. I would rather not want to catch it and let it bubble up the promise chain.